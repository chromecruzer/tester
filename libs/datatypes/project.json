{
  "name": "datatypes",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/datatypes/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/datatypes/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/datatypes"],
      "options": {
        "jestConfig": "libs/datatypes/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
