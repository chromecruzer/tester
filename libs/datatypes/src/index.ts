export * from './datatypes';
export * from './IolRecord';
export * from './CustomerRecord';
export * from './CustomerContactRecord';
export * from './ProductRecord';
export * from './EmployeeRecord';        // added by shyam for testing 
export * from './EmployeeRecordDataMappings'   //
export * from './UserRecord1'  //
export * from './UserRecordDM' // 
export * from './ConsignmentRecord';
export * from './CustomerDataMapping';
export * from './CustomerContactRecord';
export * from './CustomerContactDataMapping';
export * from './IolDataMapping';
export * from './notifications';
export * from './ProductDataMapping';
export * from './SalesRepDataMapping';
export * from './SalesRepRecord';
export * from './SalesMappingSquared';
export * from './SalesMappingRecord';
export * from './SalesJumpRecord';
export * from './UploadHeaders';
export * from './AuditRecord';
export * from './SearchSignature';
export * from './EmailSettings';
//export * from './UserRecord';
import BuildUrl from "./BuildUrl";
import {ExpirationCalculations} from "./ExpirationCalculations";
import ItemMatchStatus from './ItemMatchStatus';
import SearchSignature from "./SearchSignature";
export {BuildUrl, ExpirationCalculations, ItemMatchStatus, SearchSignature};
