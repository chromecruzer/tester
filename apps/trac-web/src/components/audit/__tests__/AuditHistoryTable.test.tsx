import {AuditHistory} from "../AuditHistory";
import {render, screen} from "@testing-library/react";
import * as reactRedux from "../../../redux/hooks";
import {AuditStatusType} from "@trac/datatypes";
import * as reactRouter from "react-router";
import {useState as useStateMock} from "react";
import userEvent from "@testing-library/user-event";

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn()
}))
describe('AuditHistoryTable', () => {
  let openAudits, closedAudits, dispatchSpy, navigateSpy, setAuditStatusSpy;
  const renderAuditHistoryTable = () => {
    render(<AuditHistory/>)
  }
  const setupSelector = (type = null as AuditStatusType) => {
    let expected;
    switch (type) {
      case 'OPEN':
        expected = openAudits;
        break;
      case 'CLOSED':
        expected = closedAudits;
        break;
      default:
        expected = [...openAudits, ...closedAudits];
        break;
    }
    jest.spyOn(reactRedux, 'useAppSelector').mockImplementation(() => ({
      audits: expected,
      type: 'locations'
    }));
  }
  beforeEach(() => {
    dispatchSpy = jest.fn();
    navigateSpy = jest.fn();
    setAuditStatusSpy = jest.fn();
    jest.spyOn(reactRedux, 'useAppDispatch').mockImplementation(() => dispatchSpy);
    jest.spyOn(reactRouter, 'useNavigate').mockImplementation(() => navigateSpy);
    (useStateMock as jest.Mock)
      .mockImplementation(() => ([null, setAuditStatusSpy]));
    openAudits = [
      {
        uuid: 'openuuid1',
        received_date: '07/11/2022',
        location_code: 'location1',
        location: 'Oh No I am under audit',
        scan_date: '06/01/2022',
        close_date: null,
        status: 'OPEN',
        auditor: 'Becky Counter',
        scanner: 'Mike Scanner'
      },
      {
        uuid: 'openuuid2',
        received_date: '07/03/2022',
        location_code: 'location13',
        location: 'I got nuttin to hide',
        scan_date: '05/23/2022',
        close_date: null,
        status: 'OPEN',
        auditor: 'Becky Counter',
        scanner: 'Bubba Scanner'
      },
    ]
    closedAudits = [
      {
        uuid: 'closeduuid3',
        received_date: '05/11/2021',
        location_code: 'location1',
        location: 'Oh No I am under audit',
        scan_date: '03/01/2021',
        close_date: '08/17/2021',
        status: 'CLOSED',
        auditor: 'Priscilla Counter',
        scanner: 'Mike Scanner'
      },
      {
        uuid: 'closeduuid4',
        received_date: '05/12/2021',
        location_code: 'location13',
        location: 'I got nuttin to hide',
        scan_date: '03/06/2021',
        close_date: '08/19/2021',
        status: 'CLOSED',
        auditor: 'Priscilla Counter',
        scanner: 'Bubba Scanner'
      },
      {
        uuid: 'closeduuid5',
        received_date: '05/10/2021',
        location_code: 'location2',
        location: 'My books are clean',
        scan_date: '02/14/2021',
        close_date: '06/17/2021',
        status: 'CLOSED',
        auditor: 'Priscilla Counter',
        scanner: 'Mary Lou Scanner'
      },
    ]
  });
  it('should display history data correctly', () => {
    setupSelector();
    renderAuditHistoryTable();
    const rows = screen.queryAllByRole('row');
    expect(rows).toMatchSnapshot();
  });
  it('should navigate to the audit uuid', () => {
    userEvent.setup();
    setupSelector();
    renderAuditHistoryTable();
    // screen.debug();
    const rows = screen.queryAllByRole('row');
    userEvent.click(rows[5]).then(() => { //TODO this is not triggering
      expect(navigateSpy.mock.calls).toMatchSnapshot();
    })
  });
  it('should filter OPEN items', () => {
    setupSelector('OPEN');
    renderAuditHistoryTable();
    const rows = screen.queryAllByRole('row');
    expect(rows).toMatchSnapshot();
  });
  it('should filter CLOSED items', () => {
    setupSelector('CLOSED');
    renderAuditHistoryTable();
    const rows = screen.queryAllByRole('row');
    expect(rows).toMatchSnapshot();
  });
});
