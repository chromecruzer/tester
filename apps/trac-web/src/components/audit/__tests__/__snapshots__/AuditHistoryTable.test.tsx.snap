// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenAudits should display history data correctly 1`] = `
Array [
  <tr
    class="bg-bl-consumer-light"
    role="row"
  >
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Location
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Status
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Location Code
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Scan Date
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Close Date
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Auditor
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Scanner
      </span>
    </th>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      Oh No I am under audit
    </td>
    <td
      class="  "
      role="cell"
    >
      OPEN
    </td>
    <td
      class="  "
      role="cell"
    >
      location1
    </td>
    <td
      class="  "
      role="cell"
    >
      06/01/2022
    </td>
    <td
      class="  "
      role="cell"
    />
    <td
      class="  "
      role="cell"
    >
      Becky Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Mike Scanner
    </td>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      I got nuttin to hide
    </td>
    <td
      class="  "
      role="cell"
    >
      OPEN
    </td>
    <td
      class="  "
      role="cell"
    >
      location13
    </td>
    <td
      class="  "
      role="cell"
    >
      05/23/2022
    </td>
    <td
      class="  "
      role="cell"
    />
    <td
      class="  "
      role="cell"
    >
      Becky Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Bubba Scanner
    </td>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      Oh No I am under audit
    </td>
    <td
      class="  "
      role="cell"
    >
      CLOSED
    </td>
    <td
      class="  "
      role="cell"
    >
      location1
    </td>
    <td
      class="  "
      role="cell"
    >
      03/01/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      08/17/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      Priscilla Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Mike Scanner
    </td>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      I got nuttin to hide
    </td>
    <td
      class="  "
      role="cell"
    >
      CLOSED
    </td>
    <td
      class="  "
      role="cell"
    >
      location13
    </td>
    <td
      class="  "
      role="cell"
    >
      03/06/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      08/19/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      Priscilla Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Bubba Scanner
    </td>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      My books are clean
    </td>
    <td
      class="  "
      role="cell"
    >
      CLOSED
    </td>
    <td
      class="  "
      role="cell"
    >
      location2
    </td>
    <td
      class="  "
      role="cell"
    >
      02/14/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      06/17/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      Priscilla Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Mary Lou Scanner
    </td>
  </tr>,
]
`;

exports[`OpenAudits should filter CLOSED items 1`] = `
Array [
  <tr
    class="bg-bl-consumer-light"
    role="row"
  >
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Location
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Status
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Location Code
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Scan Date
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Close Date
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Auditor
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Scanner
      </span>
    </th>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      Oh No I am under audit
    </td>
    <td
      class="  "
      role="cell"
    >
      CLOSED
    </td>
    <td
      class="  "
      role="cell"
    >
      location1
    </td>
    <td
      class="  "
      role="cell"
    >
      03/01/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      08/17/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      Priscilla Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Mike Scanner
    </td>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      I got nuttin to hide
    </td>
    <td
      class="  "
      role="cell"
    >
      CLOSED
    </td>
    <td
      class="  "
      role="cell"
    >
      location13
    </td>
    <td
      class="  "
      role="cell"
    >
      03/06/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      08/19/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      Priscilla Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Bubba Scanner
    </td>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      My books are clean
    </td>
    <td
      class="  "
      role="cell"
    >
      CLOSED
    </td>
    <td
      class="  "
      role="cell"
    >
      location2
    </td>
    <td
      class="  "
      role="cell"
    >
      02/14/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      06/17/2021
    </td>
    <td
      class="  "
      role="cell"
    >
      Priscilla Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Mary Lou Scanner
    </td>
  </tr>,
]
`;

exports[`OpenAudits should filter OPEN items 1`] = `
Array [
  <tr
    class="bg-bl-consumer-light"
    role="row"
  >
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Location
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Status
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Location Code
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Scan Date
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Close Date
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Auditor
      </span>
    </th>
    <th
      colspan="1"
      role="columnheader"
      style="cursor: pointer;"
      title="Toggle SortBy"
    >
      <span
        class="flex items-center px-1"
      >
        Scanner
      </span>
    </th>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      Oh No I am under audit
    </td>
    <td
      class="  "
      role="cell"
    >
      OPEN
    </td>
    <td
      class="  "
      role="cell"
    >
      location1
    </td>
    <td
      class="  "
      role="cell"
    >
      06/01/2022
    </td>
    <td
      class="  "
      role="cell"
    />
    <td
      class="  "
      role="cell"
    >
      Becky Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Mike Scanner
    </td>
  </tr>,
  <tr
    class=" border-2 border-gray-100 cursor-pointer"
    role="row"
  >
    <td
      class="  "
      role="cell"
    >
      I got nuttin to hide
    </td>
    <td
      class="  "
      role="cell"
    >
      OPEN
    </td>
    <td
      class="  "
      role="cell"
    >
      location13
    </td>
    <td
      class="  "
      role="cell"
    >
      05/23/2022
    </td>
    <td
      class="  "
      role="cell"
    />
    <td
      class="  "
      role="cell"
    >
      Becky Counter
    </td>
    <td
      class="  "
      role="cell"
    >
      Bubba Scanner
    </td>
  </tr>,
]
`;
